SETUP.md

```diff
@@ -1,330 +1,406 @@
 ```
 
-### Build the protobufs
+#### Production Setup
 
-The gRPC services are defined in the backend_service.proto. Build these with the following command:
+For production environments with domain names:
 
-```shell
-./compile_proto.sh
+```bash
+# First, configure Cloudflare API credentials in .env.backend
+nano .env.backend
 ```
 
-There is no output on the terminal unless something went wrong.
-
-### Build the backend
+Add these production settings:
+```env
+# Cloudflare API Configuration (for Let's Encrypt certificates)
+CLOUDFLARE_API_TOKEN=your-cloudflare-api-token
+LOPHIID_CERT_DOMAIN=your-domain.com
+LOPHIID_CERT_EMAIL=your-email@domain.com
+```
 
-To build the backend, run the following command:
+Then generate certificates with production mode:
+```bash
+# Generate production certificates with Let's Encrypt
+./scripts/setup-backend-certs.sh your-domain.com production
 
-```shell
-bazel build //cmd/backend:backend
+# Example:
+./scripts/setup-backend-certs.sh honeypot.mydomain.com production
 ```
 
-This will take care of downloading all the remaining dependencies and build the backend.
+**Cloudflare Setup Requirements:**
+1. Domain managed by Cloudflare DNS
+2. Cloudflare API token with Zone:Read and Zone:Edit permissions
+3. Email address for Let's Encrypt registration
 
-If you are unfamilair with bazel, you can find the binary at ./bazel-bin/cmd/backend/backend_/backend
+The script creates:
+- CA certificate in `certs/ca/ca-cert.pem` (for internal gRPC)
+- Server certificates in `certs/server/` (for internal gRPC)
+- Let's Encrypt certificates in `certs/live/lophiid-backend/` (for public HTTPS)
 
-### Build the agent
-First make sure that libmagic is installed:
+**Certificate Usage:**
+- **Backend/API HTTPS**: Uses Let's Encrypt certificates via nginx reverse proxy
+- **Agent HTTPS honeypots**: Automatically use Let's Encrypt when agent has domain name
+- **Internal gRPC**: Uses self-signed certificates (backend ↔ agent communication)
 
-```shell
-sudo apt-get install libmagic-dev
-```
+### Step 2: Configure Environment Variables
 
-Build the agent using the following command:
+**CRITICAL**: Edit `.env.backend` before starting services:
 
-```shell
-bazel build //cmd/agent:client
+```bash
+# Edit backend configuration
+nano .env.backend
 ```
 
-# Create a CA and certificates
+**Required changes:**
+```env
+# Change default passwords
+POSTGRES_PASSWORD=your-secure-password
+LOPHIID_API_KEY=your-secure-api-key
 
-SSL is used for encryption and authentication (client certs). Using this is totally optional, you can leave the ssl_cert configuration options out of the config but that is not recommended.
+# Update shared database URL with new password (used by all services)
+LOPHIID_DATABASE_URL=postgres://lophiid:your-secure-password@postgres:5432/lophiid
 
-Below are example commands for creating certificates.
+# IMPORTANT: Set UI access address (replace with your server IP/domain)
+VUE_APP_BACKEND_ADDRESS=http://192.168.1.100:9999
```

Comment:
This could also be LOPHIID_API_BACKEND_ADDRESS. Maybe change it to that or change LOPHIID_API_KEY to VUE_APP_BACKEND_API_KEY ?